packages/kit/src/runtime/client/client.js

```diff
@@ -244,16 +244,26 @@ export async function start(_app, _target, hydrate) {
 		current_history_index = current_navigation_index = Date.now();
 
 		// create initial history entry, so we can return here
-		original_replace_state.call(
-			history,
-			{
-				...history.state,
-				[HISTORY_INDEX]: current_history_index,
-				[NAVIGATION_INDEX]: current_navigation_index
-			},
-			'',
-			location.href
-		);
+		try {
+			original_replace_state.call(
+				history,
+				{
+					...history.state,
+					[HISTORY_INDEX]: current_history_index,
+					[NAVIGATION_INDEX]: current_navigation_index
+				},
+				''
+			);
+		} catch (error) {
+			// detect if the issue has been created by basic auth credentials in the current URL
+			// https://github.com/sveltejs/kit/issues/10522
+			// if so, refresh the page without credentials
+			const url = new URL(document.URL);
+			if (!(url.username || url.password)) {
+				throw error;
+			}
+			location.href = `${location.href}`;
+		}
```

Comment:
we should remove the try-catch, and add the URL check (`document.URL !== location.href` _should_ work, I think) at the top of `start`

```suggestion
		original_replace_state.call(
			history,
			{
				...history.state,
				[HISTORY_INDEX]: current_history_index,
				[NAVIGATION_INDEX]: current_navigation_index
			},
			''
		);
```

Reply:
Should we keep the `(url.username || url.password)` check, just in case we are missing another reason why `location.href` and `document.URL`? I really can't think of any other scenario, but I have to admit my knowledge of the spec is limited. It would result in something like

```js
if (document.URL !== location.href) {
    const url = new URL(document.URL);
    if (url.username || url.password) {
        location.href = `${location.href}`;
    }
}
```

Reply:
My inclination in these cases is to do the simpler thing that we're _fairly_ sure is correct, rather than the more defensive thing. If it turns out there are other cases where `document.URL` and `location.href` differ, we'll find out about them soon enough, but if the defensive code is unnecessary we'll never find out — we'll just ship extra code to every SvelteKit app user for no good reason